### ISSUE TITLE
Including 'mailgun-ruby' causes action_mailer configs in an initializer to not be used

### ISSUE THREAD

**Issue by eebs** [mailgun/mailgun-ruby#86]
Hello, I have encountered an odd scenario where I am unable to set action_mailer configuration options if I include the `mailgun-ruby` gem.

I have a Rails 4.2.7.1 app, where I configure ActionMailer in an initializer. Some of the options I set are the `default_url_options` and `asset_host`, so that images in our emails have an absolute URL rather than a relative one. Without requiring `mailgun-ruby`, emails are generated correctly. As soon as I add `gem 'mailgun-ruby'` to my Gemfile, these options are no longer present at the time the email is rendered. This causes an exception to be throw by Rails stating that the `default_url_options` are not set.

I did some digging, and found that the root cause is the `Railgun::Railtie`, which adds the `mailgun` delivery method in `before_initialize` hook. I can not find any explanation either in Rails or elsewhere that would explain why adding a delivery method would cause subsequent configuration to not be used. I looked at the [Rails initialization events](http://guides.rubyonrails.org/configuring.html#initialization-events) and found nothing that points to a reason why.

I am able to work around this issue by requiring all necessary files except the Railtie. I'm not actually using the `mailgun` delivery method, my use case requires me to manually call `Railgun.transform_for_mailgun(mail)` and `client.send_message('my-domain', message_payload)`, so adding the following to my Gemfile works:

```ruby
gem 'mailgun-ruby', '~>1.1.4', require: ['mailgun', 'railgun/attachment', 'railgun/errors', 'railgun/mailer', 'railgun/message']
```

Do you have any thoughts on why this might be happening?

For development, my initializer is effectively this:

```ruby
# config/initializers/action_mailer.rb
Rails.application.config.action_mailer.default_url_options = { host: 'localhost', :port: 3000, protocol: 'http' }
Rails.application.config.action_mailer.asset_host = 'http://localhost:3000'
Rails.application.config.action_mailer.delivery_method = :smtp
Rails.application.config.action_mailer.smtp_settings = {
  address: 'smtp.mailgun.org',
  port: 587,
  domain: 'my-domain',
  user_name: 'username',
  password: 'supersecret',
  authentication: 'plain',
  enable_starttls_auto: true,
}
Rails.application.config.action_mailer.perform_deliveries = false
```

**Comment by pirogoeth** [comment]
Hey there! It looks like this is happening because requiring `mailgun-ruby` in turn requires `railgun` when Rails is defined. The solution should be simple - remove the `railgun` require line from the `mailgun-ruby` require hook. I'll take care of this today, thanks for the report :)

**Comment by eebs** [comment]
I'm not sure that's really what you want to do. I think requiring Railgun when Rails is defined makes sense. I'm more interested in why adding a delivery method prevents subsequent configuration from being used.

**Comment by pirogoeth** [comment]
@eebs I am not entirely sure why the railtie is causing this issue. I will try to dig deeper in to it today.

**Comment by eebs** [comment]
Thanks. I'll also try and narrow things down from my end. I don't think I'm doing anything non-standard, but I'll try and create a simpler test case.

**Comment by subdigital** [comment]
I ran into this issue as well. I worked around it by moving the config to `config.action_mailer.mailgun_settings`.

**Comment by duartemvix** [comment]
I'm having something that might be related to this issue. I couldn't send e-mails using the API credentials just like instructions say so.

I was getting this after calling deliver_now on an instance of Mailer

`Railgun::ConfigurationError: Config requires `api_key` key
from /home/matt/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/mailgun-ruby-1.1.11/lib/railgun/mailer.rb:27:in `block in initialize'`

But my configurations were all set

![image](https://user-images.githubusercontent.com/19348306/50622038-869f3a00-0ef1-11e9-8045-0b634a5ee1d6.png)

![image](https://user-images.githubusercontent.com/19348306/50622048-99197380-0ef1-11e9-87ec-86e3bf38489e.png)

**Comment by bjminihan** [comment]
I've attempted the above with Devise and haven't been able to get the above working. Any advice on routing Devise mail through Mailgun?

**Comment by duartemvix** [comment]
> I've attempted the above with Devise and haven't been able to get the above working. Any advice on routing Devise mail through Mailgun?

I have managed to use Mailgun credentials for Devise to use them. I just had to be sure to check Devise default mailer configuration. If you configure it to what Mailer suits your needs, Devise willl use it.
![image](https://user-images.githubusercontent.com/19348306/50835558-24559780-133e-11e9-8e13-0c942241fdfb.png)

Make sure to have this in your environment(development, or production).rb
![image](https://user-images.githubusercontent.com/19348306/50835589-418a6600-133e-11e9-8e20-2aafdd3b52e2.png)
